athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/glue/GlueFieldLexer.java

```diff
@@ -94,8 +94,14 @@ private static Field lexComplex(String name, GlueTypeParser.Token startToken, Gl
         }
         else if (startToken.getValue().toLowerCase().equals(LIST)) {
             GlueTypeParser.Token arrayType = parser.next();
-            return FieldBuilder.newBuilder(name, Types.MinorType.LIST.getType())
-                    .addField(mapper.getField(name, arrayType.getValue())).build();
+            Field child;
+            if (arrayType.getValue().toLowerCase().equals(STRUCT)) {
+                child = lexComplex(name, arrayType, parser, mapper);
```

Comment:
can you also account for nested lists (list of lists)?

Reply:
Added support for nested lists.